pages/onboarding/index.tsx

```diff
@@ -1,15 +1,131 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState, useEffect } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface Props {
+  user: UserProfile | null;
+}
+
+const OnboardingPage: NextPage<Props> = ({ user }: Props) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: user?.bio,
+    twitter_username: user?.twitter_username,
+    linkedin_url: user?.linkedin_url,
+    github_username: user?.github_username,
+    website: user?.website,
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFormFilled, setIsFormFilled] = useState(false);
+
+  const router = useRouter();
+
+  useEffect(() => {
+    if (!user) {
+      router.push('/');
+    }
+  }, [user]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+    const isFormFilled = Object.values(formData).some(value => !!value);
+    setIsFormFilled(isFormFilled);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      setIsLoading(true);
+      await axios.patch(`/api/users/${user?.id}`, formData);
+      setIsLoading(false);
+      router.push('/directory');
+    } catch (error) {
+      setIsLoading(false);
+      console.error(error);
+    }
+  };
+
+  const isSubmitDisabled = !isFormFilled && !isLoading;
 
-const Onboarding: NextPage = () => {
   return (
     <Container maxWidth="lg" className="flex justify-between pt-4">
       <Typography variant="h1" className="text-3xl">
         Onboarding
       </Typography>
+      <form onSubmit={handleSubmit}>
```

Comment:
Can we move the form out of the header of the page? See `pages/profile/[id].tsx` for reference.

Reply:
is there a benefit to doing this?

Reply:
A form like this shouldn't be in the header... making this change now will simplify the tasks later when we add proper page layouts to the application.

Reply:
i also don't think we need a heading.  i.e. we don't need a title for the page that says "onboarding"... onboarding is mostly an internal concept.  we should be communicating this to our users in a more user-friendly way, like "Complete your profile to connect with other members!"

Reply:
@jmoldyvan Tim merged #95 which gives us proper layout components. When you merge changes from `main`, this file can be simplified.